Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/DS_poller.py

```diff
@@ -0,0 +1,171 @@
+        logger.info("got inside the poller code")
+        self.event = self.date.get_last_event(historical_days)
+        if isinstance(self.event, tuple):
+            self.after_time = self.event[0]
+            self.before_time = self.event[1]
+            logger.info("From time: %s", self.after_time)
+            logger.info("to time: %s", self.before_time)
+        else:
+            logger.info("Polling from event number " + str(self.event))
+
+    def parse_desc(self, data):
+        """
+            adds more newlines to description for good display on Azure sentinel incidents
+        """
+        arr = data.splitlines()
+        res = ""
+        for e in arr:
+            res = res + e + "\n\n"
+        return res
+
+
+    def post_azure(self, alerts_and_incidents, triage_items, app):
+        """
+            posts to azure after appending triage information on it
+        """
+
+        data = {}
+        for item in triage_items:
+            # Getting all the present triage items into the data dict
+            if 'alert-id' in item['source'] and item['source']['alert-id'] is not None:
+                data[item['source']['alert-id']] = [item, None]
+            elif 'incident-id' in item['source'] and item['source']['incident-id'] is not None:
+                data[item['source']['incident-id']] = [item, None]
+            else:
+                raise Exception(f'Triage item missing expected source ID field: {item}')
+
+        for item in alerts_and_incidents:
+            # Replacing None in the list with the incident/alert corresponding to respective triage item. 
+            if item['id'] in data:
+                data[item['id']][1] = item
+            else:
+                raise Exception(f'No matching triage item found for alert/incident: {item}')
+
+        for triage_item, alert_or_incident in data.values():
+            # validate we actually have an alert/incident before proceeding
+            if not alert_or_incident:
+                raise Exception(f'Triage item had no matching alert/incident data: {triage_item}')
+            alert_or_incident['status'] = triage_item['state']
+            alert_or_incident['triage_id'] = triage_item['id']
+            alert_or_incident['triage_raised_time'] = triage_item['raised']
+            alert_or_incident['triage_updated_time'] = triage_item['updated']
+            
+            #creating a custom json data to post into azure
+            azure_obj = {
+                **alert_or_incident,
+                'status': triage_item['state'],
+                'triage_id': triage_item['id'],
+                'triage_raised_time': triage_item['raised'],
+                'triage_updated_time': triage_item['updated'],
+                'comments': [],
+                'app': app
+            }
+
+            comment_data = self.DS_obj.get_triage_comments(triage_item['id'])
+            
+            for comment in comment_data:
+                if not comment['content']:

```

Comment:
```suggestion
                if not 'content' in comment:
```

the current version of this will error, rather than fail, if the 'content' key is missing from the `comment` object.